Typescript typings: incorrect type for ResponseData.statusCode

### What is your Test Scenario?
Create a typescript test where statusCode is checked against a string eg. '200'
 
### What is the Current behavior?
The typescript compiler will show no Error as ResponseData.statusCode is expected to be a string as typed in testcafe's index.d.ts typings
As the actual runtime value is a number, these tests always fail
 
### What is the Expected behavior?
ResponseData.statusCode should be typed as a number
 
### What is your web application and your TestCafe test code?
See sample below

<details>
<summary>Your complete test code (or attach your test files):</summary>

<!-- Paste your test code here: --> 
```
 
import { Selector, RequestLogger, ClientFunction, Role } from 'testcafe'


const testUrl = 'https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/browsers/browser-support.html'

fixture`test url for 200 statusCode as number and as string`.page(testUrl)

test('test url for 200 statusCode using expect inline check', async t => {
    const genericLogger = RequestLogger(testUrl)
    await t.addRequestHooks(genericLogger)
        .navigateTo(testUrl)
        .expect(genericLogger.contains((r:Request) => r.response.statusCode === '200'))
        .ok('This should pass if statusCode is indeed a string, but fails as it is checked against the number 200 ')
})
```
</details>

<details>
<summary>Your complete test report:</summary>

<!-- Paste your complete result test report here (even if it is huge): -->
```
arend@vn7:testcafeTypingsIssue$ npx testcafe chrome myTest.ts
 Running tests in:
 - Chrome 74.0.3729 / Linux 0.0.0

 test url for 200 statusCode as number and as string
(node:4766) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
 ✖ test url for 200 statusCode using expect inline check

   1) AssertionError: This should pass if statusCode is indeed a string, but fails as it is checked against the number 200 : expected false to be truthy

      Browser: Chrome 74.0.3729 / Linux 0.0.0

          9 |test('test url for 200 statusCode using expect inline check', async t => {
         10 |    const genericLogger = RequestLogger(testUrl)
         11 |    await t.addRequestHooks(genericLogger)
         12 |        .navigateTo(testUrl)
         13 |        .expect(genericLogger.contains((r:Request) => r.response.statusCode === '200'))
       > 14 |        .ok('This should pass if statusCode is indeed a string, but fails as it is checked against the number 200 ')
         15 |})

         at Object.<anonymous> (/home/arend/Documents/dev/playground/testcafeTypingsIssue/myTest.ts:14:10)
         at <anonymous> (/home/arend/Documents/dev/playground/testcafeTypingsIssue/myTest.ts:7:71)
         at __awaiter (/home/arend/Documents/dev/playground/testcafeTypingsIssue/myTest.ts:3:12)
         at test (/home/arend/Documents/dev/playground/testcafeTypingsIssue/myTest.ts:9:73)
         at markeredfn (/home/arend/Documents/dev/playground/testcafeTypingsIssue/node_modules/testcafe/src/api/wrap-test-function.js:17:28)
         at <anonymous> (/home/arend/Documents/dev/playground/testcafeTypingsIssue/node_modules/testcafe/src/api/wrap-test-function.js:7:5)
         at fn (/home/arend/Documents/dev/playground/testcafeTypingsIssue/node_modules/testcafe/src/test-run/index.js:242:19)
         at TestRun._executeTestFn (/home/arend/Documents/dev/playground/testcafeTypingsIssue/node_modules/testcafe/src/test-run/index.js:238:38)
         at _executeTestFn (/home/arend/Documents/dev/playground/testcafeTypingsIssue/node_modules/testcafe/src/test-run/index.js:291:24)



 1/1 failed (5s)
arend@vn7:testcafeTypingsIssue$ 

```
</details>

<details>
<summary>Screenshots:</summary>
<!-- If applicable, add screenshots to help explain the issue. -->

```

```
</details>

### Steps to Reproduce:
<!-- Describe what we should do to reproduce the behavior you encountered. -->

1. save the sample code in a file called myTest.ts
3. Execute this command ``` testcafe chrome myTest.ts ```
4. See the error...
 
### Your Environment details:
 
* testcafe version:                   1.1.4
* node.js version:                   v11.1.0
* command-line arguments:  chrome myTest.ts
* browser name and version: Version 74.0.3729.157 (Official Build) (64-bit)

* platform and version:         Linux vn7 4.20.0-042000-generic #201812232030 SMP Mon Dec 24 01:32:58 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux / Ubuntu 18.10

--- a/ts-defs/index.d.ts
+++ b/ts-defs/index.d.ts
@@ -876,12 +876,12 @@ interface RequestLogger {
      * Returns whether the logger contains a request that matches the predicate.
      * @param predicate - The predicate
      */
-    contains(predicate: Function): Promise<boolean>;
+    contains(predicate: (request: LoggedRequest) => boolean): Promise<boolean>;
     /**
      * Returns the number of requests that match the predicate.
      * @param predicate - The predicate
      */
-    count(predicate: Function): Promise<number>;
+    count(predicate: (request: LoggedRequest) => boolean): Promise<number>;
     /**
      * Clears all logged requests.
      */
@@ -889,7 +889,7 @@ interface RequestLogger {
     /**
      * Returns an array of logged requests.
      */
-    requests: Array<Request>;
+    requests: Array<LoggedRequest>;
 }
 
 interface RequestLoggerOptions {
@@ -934,7 +934,7 @@ interface RequestMock {
     respond(body?: object | string | ((req: any, res: any) => any), statusCode?: number, headers?: object): RequestMock;
 }
 
-interface Request {
+interface LoggedRequest {
     /**
      * The user agent that sent the request.
      */
@@ -973,7 +973,7 @@ interface ResponseData {
     /**
      * The status code received in the response.
      */
-    statusCode: string;
+    statusCode: number;
     /**
      * Response headers in the property-value form. Logged if the `logResponseHeaders` option is set to true.
      */

